### ISSUE TITLE
Feature: Specify date format used for parsing

### ISSUE THREAD

**Issue by JanPetterMG** [simplepie/simplepie#475]
Some feeds are using non-standard date formats.
For example, today's date could be specified as both `17-10-16` (`d-m-y`) and `16-10-17` (`y-m-d`), depending on the host.

How to specify a custom date format as source/input format?

Currently I'm handling this issue externally, re-parsing the dates for specific feeds, but when source date includes months and weekdays in a local language, everything gets a lot more complicated...

**EDIT:**
There is no way doing this in the current version (1.4.x), so consider this an feature request.
Simplepie already handles translations internally, so I guess it makes more sense to implement it, than handling the whole date-parsing process externally.

**Comment by skyzyx** [comment]
IIRC, I believe that the `get_date()` and `get_gmdate()` functions support a `$format` parameter for this very use-case.

Am I misremembering this?

**Comment by JanPetterMG** [comment]
As far as I can see, the `$format` parameter is the format of the outputted date string.
http://php.net/manual/en/function.date.php
In other words, you can specify the output format, but not the input format.

This issue isn't unfixable, adding a second input parameter to `get_date()` and `get_gmdate()` as well as some black magic to the `SimplePie_Parse_Date` class.
I don't think we'll ever see this in `v1.x`, for variety of reasons, but I might put down some work if a `v2.0-dev` branch eventually shows up...

**Comment by skyzyx** [comment]
Ahhhhh. This is actually a really good idea.

This was one of those weird edge cases that we spent a couple of weeks trying to resolve early-on (2006?) before we just did a best effort and called it a day. [Bloglines](https://en.wikipedia.org/wiki/Bloglines) (remember them?) would use non-english words for RFC 822/2822 dates when the rest of the feed was in a non-english language, which is what prompted the work.

Allowing users to write code to teach SimplePie how to parse a weird date format is an excellent idea.

https://github.com/simplepie/simplepie-ng/issues/21
